experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -202,49 +204,146 @@ export function createObjectProxy<TSchema extends ObjectNodeSchema, TTypes exten
 	return proxy;
 }
 
-const getField = <TTypes extends AllowedTypes>(target: object) => {
-	const treeNode = getTreeNode(target) as FieldNode<FieldNodeSchema>;
+/**
+ * Given the a list proxy, returns it's underlying LazySequence field.
+ */
+const getSequenceField = <TTypes extends AllowedTypes>(
+	list: SharedTreeList<AllowedTypes, "javaScript">,
+) => {
+	const treeNode = getTreeNode(list) as FieldNode<FieldNodeSchema>;
 	const field = treeNode.content;
 	return field as LazySequence<TTypes>;
 };
 
-// TODO: Experiment with alternative dispatch methods to see if we can improve performance.
-
-// For brevity, the current implementation dynamically builds a property descriptor map from a list of
-// functions we want to re-expose via the proxy.
+// Converts a proxy union to contextually typed data, extracting factory content if necessary.
+const asContextuallyTypedData = (value: ProxyNodeUnion<AllowedTypes, "javaScript">) =>
+	(value === null || typeof value !== "object"
+		? value // Return primitives as-is
+		: getFactoryContent(value) ?? value) as ContextuallyTypedNodeData; // Otherwise extract factory content (if necessary).
+
+// Used by 'insert*()' APIs to converts new content (expressed as a proxy union) to contextually
+// typed data prior to forwarding to 'LazySequence.insert*()'.
+function itemsAsContextuallyTyped(
+	iterable: Iterable<ProxyNodeUnion<AllowedTypes, "javaScript">>,
+): Iterable<ContextuallyTypedNodeData> {
+	// If the iterable is not already an array, copy it into an array to use '.map()' below.
+	const asArray = Array.isArray(iterable) ? iterable : Array.from(iterable);
+
+	return asArray.map(asContextuallyTypedData);
```

Comment:
I think `Array.from` has an overload with a mapping step that you could use instead of doing the separate map here.